src/Microsoft.AspNetCore.Mvc.Razor.Precompilation/Internal/FIleProviderUtilities.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation.Internal
+{
+    public static class FileProviderUtilities
```

Comment:
We do the weird thing where we try to preserve the relative path which makes it slightly hard to express this as a generic extension method. I could call this type something different if you'd like.